### PR TITLE
Fix issue 24075 - Can't use toChars with `ushort` or `ubyte`

### PR DESCRIPTION
**Pull request by eoan-ermine**

### CODE REVIEW

**Comment by eoan-ermine** [comment]
https://issues.dlang.org/show_bug.cgi?id=24075

File: std/conv.d [hunk 1]
```diff
@@ -5870,8 +5872,12 @@ if ((radix == 2 || radix == 8 || radix == 10 || radix == 16) &&
     assert(toChars(123) == toChars(123));
 
     {
+        assert(toChars!2(cast(ubyte) 0).array == "0");
```
Comment (ntrel) [inline std/conv.d:5875]: BTW you can use e.g. `ubyte(0)` to avoid casts to built in types.

File: std/conv.d [hunk 2]
```diff
@@ -5712,17 +5712,19 @@ private auto hexStrLiteral(String)(scope String hexData)
...
 
 auto toChars(ubyte radix = 10, Char = char, LetterCase letterCase = LetterCase.lower, T)(T value)
     pure nothrow @nogc @safe
 if ((radix == 2 || radix == 8 || radix == 10 || radix == 16) &&
-    (is(immutable T == immutable uint) || is(immutable T == immutable ulong) ||
-    radix == 10 && (is(immutable T == immutable int) || is(immutable T == immutable long))))
+        (is(immutable T == immutable ubyte) || is(immutable T == immutable ushort) || is(
+        immutable T == immutable uint) || is(immutable T == immutable ulong) ||
+        radix == 10 && (is(immutable T == immutable byte) || is(immutable T == immutable short) || is(
+        immutable T == immutable int) || is(immutable T == immutable long))))
```
Comment (dkorpel) [inline std/conv.d:5727]:
```suggestion
        isIntegral!T && (radix == 10 || isUnsigned!T)
```

**Review by dkorpel — COMMENTED** [review]
I wish it would just promote the integer instead of instantiating a new `Result` struct for each integral type, but that's an enhancement for a future PR

File: std/conv.d (hunk 1 above)
Comment (eoan-ermine) [reply std/conv.d]: Thanks!

**Review by dkorpel — APPROVED** [review]

**Pull request merged**